kotlin-reference-server/src/main/kotlin/org/stellar/reference/callbacks/rate/RateService.kt

```diff
@@ -0,0 +1,231 @@
+package org.stellar.reference.callbacks.rate
+
+import java.math.BigDecimal
+import java.math.RoundingMode
+import java.text.DecimalFormat
+import java.time.Instant
+import java.time.ZoneId
+import java.time.ZonedDateTime
+import java.util.*
+import org.stellar.anchor.api.callback.GetRateRequest
+import org.stellar.anchor.api.callback.GetRateResponse
+import org.stellar.anchor.api.sep.sep38.RateFee
+import org.stellar.anchor.api.sep.sep38.RateFeeDetail
+import org.stellar.reference.callbacks.BadRequestException
+import org.stellar.reference.callbacks.NotFoundException
+import org.stellar.reference.dao.QuoteRepository
+import org.stellar.reference.model.Quote
+
+class RateService(private val quoteRepository: QuoteRepository) {
+  private val scale = 4
+  fun getRate(request: GetRateRequest): GetRateResponse {
+    val rate =
+      when {
+        request.id != null -> getRate(request.id)
+        request.sellAmount != null || request.buyAmount != null -> {
```

Comment:
❓ I'm not sure I understand why we have this condition, won't `validate_request()` catch the case where this condition is not true?

Reply:
This is to simplify the `validate_request()`. The validate request does not check the condition where both amounts are `null`. So, we need this condition outside `validate_request()` function. I think this is legit.

Reply:
Rather than have one validation check outside the request, why can't this check be the first within the `validate_request()` function?